### PR TITLE
fix: don't set `credentials=include` when in devMode

### LINKED ISSUE #348: Unable to load local `config.json` when in devMode
**Description**
As part of #333 we route requests to `config.json` through the admin API to support authenticated scenarios. This requires the `credentials: 'include'` property to be sent to include the auth cookies in the request to admin.

Since we set `access-control-allow-origin: *` on every request in the cli proxy we cannot include this header when in devMode (loading a local version of config.json) or else you will hit a CORS error `WildCardOriginNotAllowed`.

**To Reproduce**
1. Enable dev mode on a site in the sidekick settings
2. Try and load the sidekick

**Expected behavior**
Sidekick loads the local version of `config.json`

**Screenshots**
Screenshot 2023-05-10 at 8 28 12 AM

### PR DESCRIPTION
**Pull request by dylandepass**
Fixes #348 

Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues
#348 

Thanks for contributing!

### CODE REVIEW

**Comment by dylandepass** [comment]
Thanks for taking a look @tripodsan 

Yes I see what you mean. What's weird is I did test this and I was able to preview using the sidekick when in devMode with this change. Let me dig deeper and see what that was.

Alternatively, maybe we can just omit  `credentials=include` just to requests to `tools/sidekick/config.json` when in devMode.

File: src/extension/module.js [hunk 1]
```diff
@@ -431,12 +431,14 @@
...
+   * @param {SidekickConfig} config The sidekick config
    * @returns {object}
    */
-  function getAdminFetchOptions() {
+  function getAdminFetchOptions(config) {
+    const { devMode } = config;
     const opts = {
       cache: 'no-store',
-      credentials: 'include',
+      credentials: devMode ? 'omit' : 'include',
```
Comment (tripodsan) [inline src/extension/module.js:441]: I would tweak the `opts` below, when fetching the config

File: src/extension/module.js [hunk 2]
```diff
@@ -431,12 +431,14 @@
 
   /**
    * Returns the fetch options for admin requests
+   * @param {SidekickConfig} config The sidekick config
+   * @param {boolean} omitCredentials Should we omit the credentials
    * @returns {object}
    */
-  function getAdminFetchOptions() {
+  function getAdminFetchOptions(config, omitCredentials = false) {
```
[the remark this answers is not in the document]
Comment (rofe) [reply src/extension/module.js:438]: Probably a leftover from a previous version. Let’s remove it.

**Review by rofe — CHANGES_REQUESTED** [review]

**Review by tripodsan — APPROVED** [review]

**Review by rofe — APPROVED** [review]

**Pull request merged**